### ISSUE TITLE
Multitimbral channels release not working and note off being interpreted as note on.

### ISSUE THREAD

**Issue by Sequence7** [twistedelectrons/TherapSID#23]
**Describe the bug**
When you send midi notes to channel 2 , 3, or 4 Release knobs do not alter release, on 2,  3 and 4 it plays a second Note ON, on Note OFF that is audible when Release is turned up on that voice at the frequency of the last Note sent to midi channel 1. (if no midi note has been sent to channel 1 previously the release is a very low frequency)

**Steps To Reproduce**
Steps to reproduce the behavior, e.g.:
1. Send midi note to channel 2 with release knob turned up
2. Send midi note to channel 3 or 4 with release knob turned up

**Actual behavior**
Release stage never happens, on channel 2, 3 and 4 , a second note triggers when release is turned up

**Expected behavior**
Expected voice to enter release stage of envelope on note off and not trigger a second note. 

**Additional context**
Edited, channel 2 consistently does this bug also, I must have by chance played the same note on channel 1 then channel 2, when I thought channel 2 was working.

**Comment by twisted-electrons** [comment]
After exploring my guess is an issue with mono_note in voice_allocation.hpp. 
Could mono_note be confllicting between voices? Should it be an array?

**Comment by Windfisch** [comment]
I think the best solution would be to keep track of the note last emitted by any mono note tracker and to not update that variable when no tracker currently has a note / gate active.

Alternatively (equivalently), keeping track of the most recent MonoNoteTrackers that had an active note. (because for every voice, there's two trackers affecting it).

Should be a small fix in voice_state.hpp

Currently, MonoNoteTracker returns a "current note" which is used regardless of whether it's actually active.

I can have a look into it if you like

27.03.2023 19:30:22 Twisted Electrons ***@***.***>:

**Comment by twisted-electrons** [comment]
Thanks Windfisch, I'll give it another try based on your tips and cry for help if I fail :)

**Comment by twisted-electrons** [comment]
I think it's fixed in the individual_voice_fix branch !

**Comment by Sequence7** [comment]
I have tested twisted-electrons fix, and the good news is it is fixed! the issue I called cross talk between channels is something else, but with no LFO's mapped all three voices can be sequenced as expected thankyou. I have sent twisted-electrons a video showing the issue with LFO's but I will explain here also, if any LFO's are in use even with retrig off when midi sequences are sent to more than 1 voice, they start behaving erratically and disregard what they are meant to be doing. So currently you cannot use LFO's at all while sequencing the voices multitimbrally on midi channel 2, 3, 4

I think we can probably close this once Windfisch is happy with it, and I will make a seperate bug report for the LFO's
